tracing-core/src/field.rs

```diff
@@ -276,33 +277,68 @@ macro_rules! impl_values {
         )+
     }
 }
+
+macro_rules! ty_to_nonzero {
+    (u8) => {NonZeroU8};
+    (u16) => {NonZeroU16};
+    (u32) => {NonZeroU32};
+    (u64) => {NonZeroU64};
+    (u128) => {NonZeroU128};
+    (usize) => {NonZeroUsize};
+    (i8) => {NonZeroI8};
+    (i16) => {NonZeroI16};
+    (i32) => {NonZeroI32};
+    (i64) => {NonZeroI64};
+    (i128) => {NonZeroI128};
+    (isize) => {NonZeroIsize};
+}
+
+macro_rules! impl_one_value {
+    (bool, $op:expr, $record:ident) => {
+        impl_one_value!(normal, bool, $op, $record);
+    };
+    ($value_ty:tt, $op:expr, $record:ident) => {
+        impl_one_value!(normal, $value_ty, $op, $record);
+        impl_one_value!(nonzero, $value_ty, $op, $record);
+    };
+    (normal, $value_ty:tt, $op:expr, $record:ident) => {
+        impl $crate::sealed::Sealed for $value_ty {}
+        impl $crate::field::Value for $value_ty {
+            fn record(
+                &self,
+                key: &$crate::field::Field,
+                visitor: &mut dyn $crate::field::Visit,
+            ) {
+                visitor.$record(key, $op(*self))
+            }
+        }
+    };
+    (nonzero, $value_ty:tt, $op:expr, $record:ident) => {
```

Comment:
I have not been able to eliminate this duplication as one cannot pass a macro invocation as an argument to another macro. I could probably figure out some scheme where one can pass a macro to another macro, but that would get quite convoluted I think.